internal/storage/index/index.go

```diff
@@ -58,7 +63,9 @@ type index struct {
 	dependents   map[namer.ModuleID]map[namer.ModuleID]struct{}
 	dependencies map[namer.ModuleID]map[namer.ModuleID]struct{}
 	modIDToFile  map[namer.ModuleID]string
+	rootDir      string
```

Comment:
I think it would be better to store the set of `BuildOpt` here rather than just the root dir